pkg/proxy/integrations/scram/util.go

```diff
@@ -29,7 +29,11 @@ func extractClientNonce(firstMsg string) (string, error) {
 	for _, part := range parts {
 		if strings.HasPrefix(part, "r=") {
 			// Split based on "=" and get the value of "r"
-			value := strings.Split(part, "=")[1]
+			// value := strings.Split(part, "=")[1]
+			value := strings.TrimPrefix(part, "r=")
+			if value == part {
```

Comment:
Is there any chance that this case occurs..?

Reply:
Yes I have mentioned an example for this in the PR comments.

Reply:
Yeah I have seen that but that is example for difference between `strings.TrimPrefix(part, "r=")` and `strings.Split(part, "=")[1]` right ? My comment was regarding even after using TrimPrefix will `value == part` anytime ?

Reply:
That is a check, which can be true if the decoding of base64 of a wrong payload. Using this the error can be handled.

Reply:
I mean there is a `if strings.HasPrefix(part, "r=")` then we used `strings.TrimPrefix(part, "r=")` how can value == part..?